src/punycode.cr

```diff
@@ -0,0 +1,168 @@
+module Punycode
```

Comment:
Along with proper credits for punycode.js and its author if it was used as a reference / ported over? ~~And is the license compatible?~~ yes it's MIT.